library/asb_set_state.py

```diff
@@ -0,0 +1,115 @@
+#!/usr/bin/python
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.0',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = '''
+---
+module: asb_set_state
+short_description: Sets state in a ConfigMap
+description:
+     - Provides a secure method for allowing Service Bundle developers
+       to set state during a Service Bundle action.
+       Provided key, value pair is stored in a ConfigMap for later retrieval.
+author:
+    - "Red Hat, Inc."
+options:
+    key:
+        description:
+            - Unique key
+        required: true
+    value:
+        description:
+            - Literal value
+        required: true
+ env:
+        # Set via the downward API on the APB Pod
+        - name: POD_NAME
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.name
+        - name: POD_NAMESPACE
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.namespace
+'''
+
+EXAMPLES = '''
+- name: Set some example state
+  asb_set_state:
+    key: "example-key"
+    value: "some-value"
+'''
+
+import os
+
+from ansible.module_utils.basic import AnsibleModule
+try:
+    from kubernetes import client, config
+    from kubernetes.client.rest import ApiException
+    config.load_kube_config()
+    api = client.CoreV1Api()
+except Exception as error:
+    ansible_module.fail_json(
+        msg='Error attempting to load kubernetes client: {}'.format(error))
+
+
+ENV_NAME = 'POD_NAME'
+ENV_NAMESPACE = 'POD_NAMESPACE'
+
+
+def create_config_map(name, namespace, data):
+    meta = client.V1ObjectMeta(name=name)
+    body = client.V1ConfigMap(metadata=meta, data=data)
+    try:
+        res = api.create_namespaced_config_map(namespace, body)
+    except ApiException as e:
+        module.fail_json(
+            msg='Error attempting to create required state management object: {}'.format(e))
+
+
+def update_config_map(name, namespace, data):
+    body = client.V1ConfigMap(data=data)
+    try:
+        res = api.patch_namespaced_config_map(name, namespace, body)
```

Comment:
will this merge the keys or replace all values in the configmap? Not sure how good the server side logic for that is at the moment